src/MQTTClient.php

```diff
@@ -262,44 +344,61 @@ protected function performConnectionHandshake(
 
             // read and process the acknowledgement
             $acknowledgement = $this->readFromSocket(4);
-            if (ord($acknowledgement[0]) >> 4 === 2) {
-                switch ($acknowledgement[3]) {
-                    case chr(0):
+            if ($acknowledgement && (ord($acknowledgement[0]) >> 4 === 2)) {
```

Comment:
I'm not too happy with this condition. It shouldn't be `0x00000000`, but the implicit conversion to a boolean seems a bit dirty to me. Is there a reason we don't use something like `strlen($acknowledgement) === 4` here?

Reply:
It's not quite the same, but the result is the same. Therefore I'm ok if we keep it.